### PR TITLE
fix: same_on_batch behavior in _adapted_rsampling for shapes of size 1

### PR DESCRIPTION
**Pull request by fortminors**
#### Changes

`torch.repeat` is called on the final rsampled tensor, repeating only the batch dimension and just copying the rest

Fixes https://github.com/kornia/kornia/issues/2698

#### Type of change

- [x] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — CHANGES_REQUESTED** [review]
Can you expand the test set to identify this issue ?

**Comment by fortminors** [comment]
@edgarriba Done. Please, take a look at the test case

File: test/augmentation/test_augmentation.py [hunk 1]
```diff
@@ -3726,6 +3726,15 @@ def test_smoke(self, device, dtype):
         aug = RandomFisheye(center_x, center_y, gamma, p=1.0)
         assert img.shape == aug(img).shape
 
+    def test_same_on_batch(self, device, dtype):
+        torch.manual_seed(0)
+        center_x = torch.tensor([-0.3, 0.3])
```
Comment (edgarriba) [inline test/augmentation/test_augmentation.py:3731]:
```suggestion
        center_x = torch.tensor([-0.3, 0.3], device=device, dtype=dtype)
```

File: test/augmentation/test_augmentation.py [hunk 2]
```diff
@@ -3726,6 +3726,15 @@ def test_smoke(self, device, dtype):
...
+        center_y = torch.tensor([-0.3, 0.3])
```
Comment (edgarriba) [inline test/augmentation/test_augmentation.py:3732]:
```suggestion
        center_y = torch.tensor([-0.3, 0.3], device=device, dtype=dtype)
```

**Review by edgarriba — APPROVED** [review]
